libraries/AP_Scripting/examples/DroneCAN_test.lua

```diff
@@ -0,0 +1,261 @@
+--[[
+    example of creating and sending DroneCAN messages
+--]]
+---@diagnostic disable: param-type-mismatch
+
+local MAGNETICFIELDSTRENGTHHIRES_ID = 1043
+local MAGNETICFIELDSTRENGTHHIRES_SIGNATURE = uint64_t(0x3053EBE3, 0xD750286F)
+
+local RAWAIRDATA_ID = 1027
+local RAWAIRDATA_SIGNATURE = uint64_t(0xC77DF38B, 0xA122F5DA)
+
+local PARAM_GETSET_ID = 11
+local PARAM_GETSET_SIGNATURE = uint64_t(0xA7B622F9, 0x39D1A4D5)
+
+local NODESTATUS_ID = 341
+local NODESTATUS_SIGNATURE = uint64_t(0x0F0868D0, 0xC1A7C6F1)
+
+local MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+
+local TARGET_NODE = Parameter('SCR_USER1')
+
+-- a handle we will use for broadcasting
+local dc_handle = DroneCAN_Handle(0)
+
+-- a handle for fetching parameters
+local param_handle = DroneCAN_Handle(0)
+param_handle:data_type(PARAM_GETSET_ID)
+param_handle:signature(PARAM_GETSET_SIGNATURE)
+
+--[[
+    setup subscription to NodeStatus
+--]]
+local nodestatus_handle = DroneCAN_Handle(0)
+nodestatus_handle:data_type(NODESTATUS_ID)
+nodestatus_handle:signature(NODESTATUS_SIGNATURE)
+nodestatus_handle:subscribe()
+
+--[[
+    setup subscription to raw air data
+--]]
+local airspeed_handle = DroneCAN_Handle(0)
+airspeed_handle:data_type(RAWAIRDATA_ID)
+airspeed_handle:signature(RAWAIRDATA_SIGNATURE)
+airspeed_handle:subscribe()
+
+-- table of all nodes
+local node_status = {}
+
+
+--[[
+    send highres mag using a global handle
+--]]
+local function send_mag_highres()
+    dc_handle:signature(MAGNETICFIELDSTRENGTHHIRES_SIGNATURE)
+    dc_handle:data_type(MAGNETICFIELDSTRENGTHHIRES_ID)
```

Comment:
Its a little bit odd to have a handle but then change it for each send. I wonder if broadcast would be better to not go via the helper? EG:

```
DroneCAN:broadcast(0, MAGNETICFIELDSTRENGTHHIRES_SIGNATURE, MAGNETICFIELDSTRENGTHHIRES_ID, payload)
```

Or we could establish a pattern of using one helper per message type.